### PR TITLE
Fix radio button indicator position

### PR DESCRIPTION
**Pull request by patchwork-body**
[Link to updated components in Deploy Preview](https://deploy-preview-550--neo-react-library-storybook.netlify.app/?path=/story/components-radio-group--radio-group-example)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

Align the radio indicator with its label

`@avaya-dux/dux-design`: This PR Aligns the radio-buttons with the [Neo guidelines](https://design.avayacloud.com/components/radio)

- According to the design the radio buttons should look like this
Screenshot 2024-09-26 at 17 01 29
- But actually it looks like this (Production storybook):
Screenshot 2024-09-26 at 17 05 27

`@avaya-dux/dux-devs`: A few CSS rules has been added to overwrite an absolute position of the radio label:before pseudo-element

- column view
Screenshot 2024-09-26 at 17 06 40

- row view
Screenshot 2024-09-26 at 17 06 54

**EXAMPLE IMAGE(S):**
Screenshot 2024-09-26 at 16 28 13
Screenshot 2024-09-26 at 16 27 55

**EXAMPLE FUNCTIONALITY GIF:**
radio-group-alignment-pr

## Summary by CodeRabbit

- **New Features**
	- Enhanced styling for radio button elements, ensuring consistent visual representation across different states (normal, checked, disabled).
- **Bug Fixes**
	- Adjusted positioning of pseudo-elements for improved alignment with labels and tooltips.
	- Refined layout for inline radio button groups to ensure consistent spacing and alignment.
	- Improved interaction behavior with explicit pointer-events settings.

### CODE REVIEW

**Review by joe-s-avaya — APPROVED** [review]

**Comment by joe-s-avaya** [comment]
Hi @patchwork-body thanks a bunch for helping us out! Always happy to see people from other braches joining in on the fun! These PRs look great, the only thing I wanted to point out is that, by "tagged @avaya-dux/dux-devs" we mean tagged as reviewers:

Screenshot 2024-09-26 at 14 38 57

By tagging the dux-devs as reviewers we'll notice it sooner. This just a small nit, and again, thanks for helping us out!

**Review by Justin-Avaya — APPROVED** [review]

**Comment by matthewAvaya** [comment]
@patchwork-body It looks like you fixed the vertical positioning of the radio (circle) to be better aligned with the label. This looks great, and thanks for bringing this up.

**Review by matthewAvaya — APPROVED** [review]

**Comment by patchwork-body** [comment]
> Hi @patchwork-body thanks a bunch for helping us out! Always happy to see people from other braches joining in on the fun! These PRs look great, the only thing I wanted to point out is that, by "tagged @avaya-dux/dux-devs" we mean tagged as reviewers:
> 
> Screenshot 2024-09-26 at 14 38 57
> 
> By tagging the dux-devs as reviewers we'll notice it sooner. This just a small nit, and again, thanks for helping us out!

looks like, I don't have permissions for that
Screenshot 2024-09-27 at 09 33 41

File: src/components/Radio/RadioGroup_shim.css [hunk 1]
```diff
@@ -79,6 +95,14 @@
 	left: 3px;
 }
 
+.neo-radio:checked + label:after {
+	top: 3px;
+}
+
+.neo-input-group--inline .neo-radio:checked + label:after {
+	top: 5px;
```
Comment (yangAtSpoken) [inline src/components/Radio/RadioGroup_shim.css:103]: This rule is not needed as it is overwriting the same value as shown in the screenshot.
Screenshot 2024-09-27 at 11 23 16 AM

**Review by yangAtSpoken — CHANGES_REQUESTED** [review]
The fix of vertical radios are great.  However, the horizontal /inline radios are already working as expect.  The inline radios, that are not working in playground, are due to bad line-height.  The portal is probably using an outdated neo-react library.

**Comment by patchwork-body** [comment]
> The fix of vertical radios are great. However, the horizontal /inline radios are already working as expect. The inline radios, that are not working in playground, are due to line-height is too small. The winning rule has the same specificity as the one in the shim file at line 20. This might be due to the Astro is putting neo-css after neo-react. But the fix should be make the rule in shim file more specific. Screenshot 2024-09-27 at 11 37 25 AM

fascinating!, I guess this error happens only in prod? bc I don't have this problem locally 
Screenshot 2024-10-01 at 17 27 26

File: src/components/Radio/RadioGroup_shim.css [hunk 2]
```diff
@@ -68,6 +76,14 @@
 	top: 2px;
 }
 
+.neo-radio + label:before {
+	top: 0;
+}
+
+.neo-input-group--inline .neo-radio + label:before {
+	top: 2px;
```
[the remark this answers is not in the document]
Comment (patchwork-body) [reply src/components/Radio/RadioGroup_shim.css]: good catch thanks! solved in there 522add126de579b67e1e7c9e13d497f77d5576c1

**Comment by patchwork-body** [comment]
> The fix of vertical radios are great. However, the horizontal /inline radios are already working as expect. The inline radios, that are not working in playground, are due to line-height is too small. The winning rule has the same specificity as the one in the shim file at line 20. This might be due to the Astro is putting neo-css after neo-react. But the fix should be make the rule in shim file more specific. Screenshot 2024-09-27 at 11 37 25 AM

I guess it's only happening in prod? I don't have such problem locally
Screenshot 2024-10-01 at 17 27 26

Maybe we could remove this rule completely?
In my case line-height: 16px works well when top for pseudo element (label:before) is set to 0px https://github.com/avaya-dux/neo-react-library/pull/550/files#diff-6b60d961462bb26fb84f7e21aa4548624b7041ba5b4bb65fac47f89a08c0ee00R67

**Review by yangAtSpoken — APPROVED** [review]
Looks good after combining the rules.

**Comment by joe-s-avaya** [comment]
@patchwork-body I've just added you as a contributor to this repo. As soon as you accept that role you should have the ability to merge this. If you do not, please tag me in a comment and I'll merge it for you. 

And thanks for your patience in working with us, it's great to see another Avaya team helping out!

**Comment by patchwork-body** [comment]
@yangAtSpoken @joe-s-avaya can someone please merge this one already?

**Comment by yangAtSpoken** [comment]
@patchwork-body I merged it. Thx.

**Pull request merged**